tuning/trainercontroller/operations/logcontrol.py

```diff
@@ -0,0 +1,65 @@
+# Third Party
+from transformers import TrainingArguments
+from transformers.utils import logging
+
+# Local
+from .operation import Operation
+
+logger = logging.get_logger(__name__)
+logger.setLevel(level=logging.DEBUG)
+
+VALID_LOG_LEVELS = {
+    "ERROR": logging.ERROR,
+    "WARNING": logging.WARNING,
+    "INFO": logging.INFO,
+    "DEBUG": logging.DEBUG,
+}
+
+
+class LogControl(Operation):
+    """Operation that can be used to log useful information on specific events."""
+
+    def __init__(self, log_format: str, log_level: str, **kwargs):
+        """Initializes the HuggingFace controls. In this init, the fields with `should_` of the
+        transformers.TrainerControl data class are extracted, and for each of those fields, the
+        control_action() method's pointer is set, and injected as a class member function.
+
+        Args:
+            kwargs: List of arguments (key, value)-pairs
+        """
+        if log_level not in VALID_LOG_LEVELS:
+            raise ValueError(
+                "Specified log_level [%s] is invalid for LogControl" % (log_level)
+            )
+        self.log_level = VALID_LOG_LEVELS[log_level]
+        self.log_format = log_format
+        super().__init__(**kwargs)
+
+    def should_log(
+        self,
+        event_name: str,
+        tc_metrics: dict,
+        args: TrainingArguments,
+        **kwargs,
+    ):
+        """This method peeks into the stack-frame of the caller to get the action the triggered
+        a call to it. Using the name of the action, the value of the control is set.
+
+        Args:
+            control: TrainerControl. Data class for controls.
+            kwargs: List of arguments (key, value)-pairs
+        """
+        log_msg = self.log_format.format(
+            event_name=event_name,
+            tc_metrics=tc_metrics,
+            args=args,
+            **kwargs,
+        )
+        if self.log_level == logging.ERROR:
+            logger.error(log_msg)
+        elif self.log_level == logging.WARNING:
+            logger.warning(log_msg)
+        elif self.log_level == logging.INFO:
+            logger.info(log_msg)
+        elif self.log_level == logging.DEBUG:
+            logger.debug(log_msg)
```

Comment:
Can't we use a function like https://github.com/python/cpython/blob/b93386e1c4dbc871148e081ae147f7d1e8d21444/Lib/logging/__init__.py#L1594 instead?